src/components/Map.js

```diff
@@ -19,13 +25,27 @@ const Map = ({
     ).then((res) => setNearbyPlaces(res));
   }, [mapProperties.center.lat, mapProperties.center.lng, setNearbyPlaces]);
 
+  function handleApiLoaded(map, maps) {
+    setMapsObj(map, maps);
+  }
+
+  // only runs if maps object is populated
+  if (!isObjEmpty(mapsObj)) {
+    const { map } = mapsObj;
+    map.addListener("center_changed", function () {
+      setUserHasPanned(true);
+    });
+  }
+
   return (
     <div style={{ height: "calc(66.67vh - 1.25rem)", width: "100%" }}>
       <GoogleMapReact
         bootstrapURLKeys={{ key: "AIzaSyA_jF-TPUl8qTMZ3BKFTrFOolH9wR7NOz4" }}
         center={mapProperties.center}
         zoom={mapProperties.zoom}
         options={{ clickableIcons: false }}
+        yesIWantToUseGoogleMapApiInternals
+        onGoogleApiLoaded={(map, maps) => handleApiLoaded(map, maps)}
```

Comment:
I'm pretty sure you can change this line to:

    onGoogleApiLoaded={setMapsObj}

Reply:
The arguments should get passed through to the function, eliminating the need for the `handleApiLoaded` function altogether!